romancal/photom/tests/test_photom.py

```diff
@@ -0,0 +1,206 @@
+    """Create a photom table for WFI.
+
+    Parameters
+    ----------
+    min_r : float
+        Minimum value to assign when populating the photometry array.
+
+    max_r : float
+        Maximum value to assign when populating the photometry array.
+
+    Returns
+    -------
+    photom_model : Roman Photom reference datamodel
+        photom roman datamodel
+
+    """
+
+    # Keyword and eleents list for phot_table construction
+    optical_element = ["F062", "F087", "F106", "F129", "W146", "F158", "F184", "F213",
+                       "GRISM", "PRISM", "DARK"]
+    none_type_elements = ["GRISM", "PRISM", "DARK"]
+    keyword = ["photmjsr", "uncertainty", "pixelareasr"]
+    nrows = len(optical_element)
+
+    # Create sample photometry keyword values
+    photmjsr = np.linspace(min_r, min_r + (nrows - 1.) * delta, nrows) * u.megajansky / u.steradian
+    uncertainty = np.linspace(min_r/20.0, min_r/20.0 + (nrows - 1.) * delta/20.0, nrows) * \
+                  u.megajansky / u.steradian
+
+    # Create sample area keyword values
+    area_ster = 2.31307642258977E-14 * u.steradian
+    pixelareasr = np.ones(nrows, dtype=np.float32) * area_ster
+
+    # Bundle values into a list
+    values = list(zip(photmjsr, uncertainty, pixelareasr))
+
+    #Create dictionary containing all values
+    reftab = {}
+    for element_idx, element in enumerate(optical_element):
+        key_dict = {}
+        for key_idx, key in enumerate(keyword):
+            key_dict[key] = values[element_idx][key_idx]
+        # GRISM, PRISM, and DARK optical elements shuld have their
+        # photomeetry keywords set to None
+        if element in none_type_elements:
+            key_dict["photmjsr"] = None
+            key_dict["uncertainty"] = None
+        reftab[element] = key_dict
+
+    # Create default datamodel
+    photom_model = testutil.mk_wfi_img_photom()
+
+    # Copy values above into defautl datamodel
+    photom_model.phot_table=reftab
+
+    return photom_model
+
+
+def test_no_photom_match():
+    """Test apply_photom warning for no match"""
+
+    # Create sample WFI Level 2 science datamodel
+    input_model = testutil.mk_level2_image()
+
+    # Create photom reference datamodel
+    photom_model = create_photom_wfi_image(min_r=3.1, delta=0.1)
+
+    # Remove key for failed test (that won't fail validation)
+    photom_model.phot_table.pop("W146")
+
+    # Select optical element
+    input_model.meta.instrument.optical_element = "W146"
+
+    # Set bad values which would be overwritten by apply_photom
+    input_model.meta.photometry.pixelarea_steradians = -1.0 * u.sr
+    input_model.meta.photometry.conversion_megajanskys = -1.0 * u.megajansky / u.steradian
+    input_model.meta.photometry.conversion_microjanskys_uncertainty = \
+        -1.0 * u.microjansky / u.arcsecond ** 2
+
+    # Look for now non existent W146 optical element
+    output_model = photom.apply_photom(input_model, photom_model)
```

Comment:
See my question on whether we should use the builtin `warnings` module.

Reply:
Updated.